StRoot/StETofUtil/StETofGeometry.cxx

```diff
@@ -586,10 +625,10 @@ StETofGeometry::init( TGeoManager* geoManager, const double* safetyMargins )
 {
     init( geoManager, safetyMargins, false );
 }
-
+*/
 
 void
-StETofGeometry::init( TGeoManager* geoManager, const double* safetyMargins, const bool& useHelixSwimmer )
+StETofGeometry::init( TGeoManager* geoManager, const double safetyMargins[2] = (double[2]){ 0, 0 }, const bool& useHelixSwimmer = false )
```

Comment:
This C-cast is quite ugly and I am not even sure if it is valid to use an initializer like this in C++. Why not simply use a `nullptr` or `std::array<double, 2>`?

Reply:
Agree on the uglyness, but it was the only way i got it working with a default :) . I want to keep the c-style array notation as that is what is provided with by all classes and macros that use this class. I could provide a null pointer default, but then i would have to check for the nullptr and set the default (0,0) in the method. I considered that even more ugly.

Reply:
You can keep the pointer to pass a C-array but in C++ you can't take an address of a temporary array. It likely compiles for you because you use gcc with GNU extensions. If you add the `-pedantic -Werror` options to gcc your code won't compile. On the other hand, taking an address of a temporary object is OK in this context. So, I already mentioned two options to deal with this, another one would be to use an `initializer_list<double>{0, 0}` to create a proxy object. Here is a link if you want to play with this code https://godbolt.org/z/czPrqnf9e

By the way, this looks ugly `const bool& useHelixSwimmer = false` 😄  Anyway, if it works and nobody else cares, keep it. I feel like I already exhausted the time I want to spend reviewing this PR. It would be nice to hear from at least another reviewer but if not let's merge it already.

Reply:
Thanks for the example :)